fe/fe-core/src/main/java/org/apache/doris/alter/CloudRollupJobV2.java

```diff
@@ -0,0 +1,209 @@
+            sleepSeveralSeconds();
+            tryTimes++;
+        }
+
+        LOG.info("postProcessRollupIndex finished, dbId:{}, tableId:{}, jobId:{}, rollupIndexList:{}",
+                dbId, tableId, jobId, rollupIndexList);
+    }
+
+    @Override
+    protected void createRollupReplica() throws AlterCancelException {
+        Database db = Env.getCurrentInternalCatalog()
+                .getDbOrException(dbId, s -> new AlterCancelException("Database " + s + " does not exist"));
+
+        // 1. create rollup replicas
+        OlapTable tbl;
+        try {
+            tbl = (OlapTable) db.getTableOrMetaException(tableId, Table.TableType.OLAP);
+        } catch (MetaNotFoundException e) {
+            throw new AlterCancelException(e.getMessage());
+        }
+
+        long expiration = (createTimeMs + timeoutMs) / 1000;
+        tbl.readLock();
+        try {
+            Preconditions.checkState(tbl.getState() == OlapTableState.ROLLUP);
+            try {
+                List<Long> rollupIndexList = new ArrayList<Long>();
+                rollupIndexList.add(rollupIndexId);
+                ((CloudInternalCatalog) Env.getCurrentInternalCatalog())
+                    .prepareMaterializedIndex(tbl.getId(), rollupIndexList, expiration);
+                for (Map.Entry<Long, MaterializedIndex> entry : this.partitionIdToRollupIndex.entrySet()) {
+                    long partitionId = entry.getKey();
+                    Partition partition = tbl.getPartition(partitionId);
+                    if (partition == null) {
+                        continue;
+                    }
+                    TTabletType tabletType = tbl.getPartitionInfo().getTabletType(partitionId);
+                    MaterializedIndex rollupIndex = entry.getValue();
+                    Cloud.CreateTabletsRequest.Builder requestBuilder =
+                            Cloud.CreateTabletsRequest.newBuilder();
+                    for (Tablet rollupTablet : rollupIndex.getTablets()) {
+                        OlapFile.TabletMetaCloudPB.Builder builder =
+                                ((CloudInternalCatalog) Env.getCurrentInternalCatalog())
+                                    .createTabletMetaBuilder(tableId, rollupIndexId,
+                                    partitionId, rollupTablet, tabletType, rollupSchemaHash,
+                                    rollupKeysType, rollupShortKeyColumnCount, tbl.getCopiedBfColumns(),
+                                    tbl.getBfFpp(), null, rollupSchema,
+                                    tbl.getDataSortInfo(), tbl.getCompressionType(), tbl.getStoragePolicy(),
+                                    tbl.isInMemory(), true,
+                                    tbl.getName(), tbl.getTTLSeconds(),
+                                    tbl.getEnableUniqueKeyMergeOnWrite(), tbl.storeRowColumn(),
+                                    tbl.getBaseSchemaVersion());
+                        requestBuilder.addTabletMetas(builder);
```

Comment:
Please add a function maybe named createRollupReplicaForPartition to reduce indent.